TicTacToe2/Math/Vec2.cs

```diff
@@ -7,21 +7,86 @@ namespace TicTacToe2.Math
     class Vec2 // 2d vector
     {
         // Properties: X, Y (int). These properties should be read-only, meaning they should only have get, not set
+        public int X { get; }
+
+        public int Y { get; }
 
         // Constructor that takes x, y arguments
+        public Vec2(int x, int y)
+        {
+            X = x;
+            Y = y;
+        }
+
         // Constructor that takes no arguments and sets X and Y to zero
+        public Vec2()
+        {
+            X = 0;
+            Y = 0;
+        }
 
         // Methods to perform these operations:
         // 1. Add int value x to the X value of the vector
+        public Vec2 AddX(int x)
+        {
+            return new Vec2(X + x, Y);
+        }
+
         // 2. Add int value y to the Y value of the vector
+        public Vec2 AddY(int y)
+        {
+            return new Vec2(X, Y + y);
+        }
+
         // 3. Add int values x and y to the vector
+        public Vec2 AddXY(int x, int y)
```

Comment:
I would recommend naming this function just Add. A function is defined by both its name and argument list. Adding just one int requires a name more descriptive than "Add" - what are you adding to? But adding two ints to a 2d vector can reasonably be assumed to add to both values of the vector. Both this function and the function that adds another Vec2 can be named Add.